src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -384,6 +387,9 @@ private void CountDownChildTest()
             _childTestCountdown.Signal();
             if (_childTestCountdown.CurrentCount == 0)
             {
+                if (Test.TestType == "Theory" && Result.ResultState == ResultState.Success && Result.PassCount == 0)
```

Comment:
It bothered me writing it as well... but I think the best solution is to introduce a type of ITest that is a theory and I didn't want to mix that in with this issue.